homeassistant/components/knx/sensor.py

```diff
@@ -53,30 +50,25 @@ def _create_sensor(xknx: XKNX, config: ConfigType) -> XknxSensor:
 class KNXSensor(KnxEntity, SensorEntity):
     """Representation of a KNX sensor."""
 
+    _device: XknxSensor
+
     def __init__(self, xknx: XKNX, config: ConfigType) -> None:
         """Initialize of a KNX sensor."""
-        self._device: XknxSensor
         super().__init__(_create_sensor(xknx, config))
-        self._unique_id = f"{self._device.sensor_value.group_address_state}"
+        self._attr_device_class = (
+            self._device.ha_device_class()
+            if self._device.ha_device_class() in DEVICE_CLASSES
+            else None
+        )
```

Comment:
None is the default, we can use that.
```suggestion
        if device_class := self._device.ha_device_class():
            self._attr_device_class = device_class
```

Reply:
shall we omit the check for `in DEVICE_CLASSES` ?

Reply:
Missed that, thanks 👍